### PR TITLE
Switch in drivers to event emitter 

### PR DESCRIPTION
**Pull request by xaiki**
ok,
this is a big pull request.
but I'm trying to get mosto in shape for 17a, and there is still loads of work.
this makes the whole code base cleaner, and passes tests.

with this code in we can elegantly implement mosto.js as an event listener.

### CODE REVIEW

File: mosto.js [hunk 1]
```diff
@@ -13,24 +14,21 @@ function mosto(configFile) {
...
-        playlist.loaded = self.playlists[i].loaded;
-        self.playlists[i] = playlist;
-        console.log("mbc-mosto: [INFO] Updated playlist:\nname: " + playlist.name 
-                + "\nstartDate: " + playlist.startDate 
-                + "\nendDate: " + playlist.endDate);
+        var list = _.find(self.playlists, function (list) {
+            return list.name == playlist.name});
+
+        if (!list)
+            return self.addPlaylist (playlist);
```
Comment (Lacrymology) [inline mosto.js:23]: I only see the "add" in case the playlist doesn't exist, but I don't see an update here.

**Comment by Lacrymology** [comment]
If this passes the tests, then there's fault in my tests (actually, there's an issue about that). This doesn't break functionality, but it does break assumptions. getWindow should not persist the new window

**Comment by Lacrymology** [comment]
This patch breaks functionalty, why was it merged without my comments even being answered to?

File: drivers/playlists/mongo-driver.js [hunk 2]
```diff
@@ -131,15 +169,15 @@ function mongo_driver(conf) {
         });
     };
 
-    mongo_driver.prototype.createPlaylist = function(sched, callback) {
+    mongo_driver.prototype.createPlaylist = function(sched, method) {
```
Comment (Lacrymology) [inline drivers/playlists/mongo-driver.js:172]: For `getPlaylists` to work this needs to call a callback with `(err, result)`, this patch breaks `getPlaylists` which is the freaking most important function of the driver, and it was merged.

I **said** this before the merge, and I **said** this was untested both in the PR comments and in the SCRUMM meeting

**Comment by xaiki** [comment]
IT IS NOT OK TO MERGE A PR WITH ISSUES RAISED BY OTHERS.
reverted.

**Comment by Lacrymology** [comment]
damnit, I'd based my other PR on this now XD

I'll keep it as is for now, and we'll review it tomorrow face to face

**Pull request merged**